ci-operator/step-registry/gather/azure-resource/gather-azure-resource-commands.sh

```diff
@@ -0,0 +1,98 @@
+#!/bin/bash
+set -o nounset
+set -o errexit
+set -o pipefail
+
+function run_command() {
+    local CMD="$1"
+    echo "Running Command: ${CMD}"
+    eval "${CMD}"
+}
+
+export PATH=$PATH:/tmp
+mkdir -p /tmp
+
+echo "$(date -u --rfc-3339=seconds) - Installing tools..."
+
+# Make sure jq is installed
+if ! command -v jq; then
+    # TODO move to image
+    curl -sL https://github.com/stedolan/jq/releases/download/jq-1.6/jq-linux64 > /tmp/jq
+    chmod +x /tmp/jq
+fi
+
+function cli_Login() {
+    # set the parameters we'll need as env vars
+    AZURE_AUTH_LOCATION="${CLUSTER_PROFILE_DIR}/osServicePrincipal.json"
+    AZURE_AUTH_CLIENT_ID="$(cat ${AZURE_AUTH_LOCATION} | jq -r .clientId)"
+    AZURE_AUTH_CLIENT_SECRET="$(cat ${AZURE_AUTH_LOCATION} | jq -r .clientSecret)"
+    AZURE_AUTH_TENANT_ID="$(cat ${AZURE_AUTH_LOCATION} | jq -r .tenantId)"
+
+    # while gathering logs from a private cluster, proxy setting is required for connecting cluster
+    if test -f "${SHARED_DIR}/proxy-conf.sh"
+    then
+        echo "Going to enable client proxy by 'source ${SHARED_DIR}/proxy-conf.sh'"
+        source "${SHARED_DIR}/proxy-conf.sh"
+    else
+        echo "No '${SHARED_DIR}/proxy-conf.sh' found, skip enabling cient proxy."
+    fi
+
+    # az should already be there
+    command -v az
+    az version
+
+    # log in with az
+    if [[ "${CLUSTER_TYPE}" == "azuremag" ]]; then
+        run_command "az cloud set --name AzureUSGovernment" || return 1
+    elif [[ "${CLUSTER_TYPE}" == "azure4" ]]; then
+        run_command "az cloud set --name AzureCloud" || return 1
+    elif [[ "${CLUSTER_TYPE}" == "azurestack" ]]; then
+        echo "Unsupported cluster: ${CLUSTER_TYPE}"
+        exit 1
+    else
+        echo "Unexpected cluster: ${CLUSTER_TYPE}"
+        exit 1
+    fi
+    echo "$(date -u --rfc-3339=seconds) - Logging in to Azure..."
+    az login --service-principal -u "${AZURE_AUTH_CLIENT_ID}" -p "${AZURE_AUTH_CLIENT_SECRET}" --tenant "${AZURE_AUTH_TENANT_ID}" --output none || return 1
+}
+
+function getResouceGroup() {
+    local CLUSTER_NAME
+    CLUSTER_NAME="${NAMESPACE}-${JOB_NAME_HASH}"
+
+    run_command "az group list -o tsv | grep ${CLUSTER_NAME}"
+
+    if [[ $(az group list -o tsv | grep -c ${CLUSTER_NAME}) -gt 1 ]]; then
+        RESOURCE_GROUP=$(az group list -o tsv | grep ${CLUSTER_NAME} | grep -v "${CLUSTER_NAME}-rg" | awk '{print $4}')
+    else
+        RESOURCE_GROUP=$(az group list -o tsv | grep ${CLUSTER_NAME} | awk '{print $4}')
+    fi
+
+    run_command "az group show --name $RESOURCE_GROUP" || return 1
+    export RESOURCE_GROUP
+}
+
+function gatherLBs() {
+    run_command "az network lb list --resource-group $RESOURCE_GROUP > $OUTPUT_DIR/lbs.json"
```

Comment:
Could we make the output easier to read? `lb list` indeed could list all the  configurations of the LBs inside this resource group, what we cares about more maybe whether masters were added into the backendpool of the public LB or not.

```
[root@preserve-gpei-worker ~]# az network lb address-pool address list --lb-name ci-op-3lx2lb8l-a9642-tp7dm --resource-group ci-op-3lx2lb8l-a9642-tp7dm-rg --pool-name ci-op-3lx2lb8l-a9642-tp7dm -o table
Name                                                                            ResourceGroup
------------------------------------------------------------------------------  -----------------------------
ci-op-3lx2lb8l-a9642-tp7dm-rg_ci-op-3lx2lb8l-a9642-tp7dm-master-1-nicpipConfig  ci-op-3lx2lb8l-a9642-tp7dm-rg
ci-op-3lx2lb8l-a9642-tp7dm-rg_ci-op-3lx2lb8l-a9642-tp7dm-master-0-nicpipConfig  ci-op-3lx2lb8l-a9642-tp7dm-rg
ci-op-3lx2lb8l-a9642-tp7dm-rg_ci-op-3lx2lb8l-a9642-tp7dm-master-2-nicpipConfig  ci-op-3lx2lb8l-a9642-tp7dm-rg
18b4cb96-25ed-44c0-a4a0-b614e349cfe2                                            ci-op-3lx2lb8l-a9642-tp7dm-rg
cfd30b69-18b7-4843-9fe5-1fb34ef6d6df                                            ci-op-3lx2lb8l-a9642-tp7dm-rg
9f25d2bf-d826-4ded-b722-0c76762af481                                            ci-op-3lx2lb8l-a9642-tp7dm-rg
```

Reply:
https://gcsweb-qe-private-deck-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/qe-private-deck/pr-logs/pull/openshift_release/35435/rehearse-35435-periodic-ci-openshift-openshift-tests-private-release-4.12-amd64-nightly-azure-ipi-disconnected-fullyprivate-p2-f14/1616407979895033856/artifacts/azure-ipi-disconnected-fullyprivate-p2-f14/gather-azure-resource/artifacts/

Reply:
I think what gpei suggested is a eady-reading output instead of json format. 

Another suggestion, I suggest we just need to printed out the info, but not need to save them into artifacts dir.